tests/test_runner/tests.py

```diff
@@ -734,3 +735,49 @@ def test_run_checks_passes_and_teardown_raises(self):
                     runner.run_tests(['test_runner_apps.sample.tests_sample.TestDjangoTestCase'])
             self.assertTrue(teardown_databases.called)
             self.assertFalse(teardown_test_environment.called)
+
+
+class NoOpTestRunner(DiscoverRunner):
+    """
+    This test runner supports a stream attribute to divert unittest's
+    output to a StringIO, for example.
+    """
+
+    def __init__(self):
+        super().__init__(verbosity=0, interactive=False)
+        self.stream = StringIO()
+
+    def get_test_runner_kwargs(self):
+        kwargs = super().get_test_runner_kwargs()
+        kwargs['stream'] = self.stream
+        return kwargs
+
+    def setup_test_environment(self, **kwargs):
+        return
+
+    def setup_databases(self, **kwargs):
+        return
+
+    def run_checks(self, databases):
+        return
+
+    def teardown_databases(self, old_config, **kwargs):
+        return
+
+    def teardown_test_environment(self, **kwargs):
+        return
+
+
+class DiscoverRunnerDeprecationTests(SimpleTestCase):
+    runner = NoOpTestRunner()
```

Comment:
I'd move it to the `setUp()`:
```python
def setUp(self):
    self.runner = NoOpTestRunner(verbosity=0, interactive=False)
```

Reply:
I would also consider turning that into an instance method called something like `get_runner()` and starting each test method with `runner = self.get_runner()`. The reason is that instantiating a runner is "cheap." You also don't have to think / worry about whether the runner has state that you might unwittingly be carrying from one test to the other (e.g. attributes set when a method is executed).